### ISSUE TITLE
Index out of bound exception

### ISSUE THREAD

**Issue by dezarion** [mikepenz/FastAdapter#589]
I work on project where included FastAdapter

```
implementation('com.mikepenz:fastadapter:2.6.2@aar') {
        transitive = true
    }
    implementation 'com.mikepenz:fastadapter-extensions:2.6.3@aar'
    implementation 'com.mikepenz:fastadapter-commons:2.6.3@aar'

```
And periodically I get a same crash from users in production. I know, version is outdated, but I cannot just up it to 3.x.x, cause I cannot reproduce this crash on my devices on 3.x.x or 2.6.x to be sure that crash may be fixed by some version.

Please tell me, do you know about this problem? Is there option to fix it? Was it fixed in 3.x.x?

There is code how I Initialize adapter:
```
val adapter: ItemAdapter<BaseItem<*, *, *>> = ItemAdapter<BaseItem<*, *, *>>().apply {
    wrap(FastAdapter())
  }
```

And how I set data to adapter:

```
private fun onDataUpdated(data: List<BaseItem<*, *, *>>) {
    FastAdapterDiffUtil.set(adapter, data, BaseDiffCallback(), true)
  }
```

```
Fatal Exception: java.lang.IndexOutOfBoundsException: Invalid index 25, size is 25
       at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
       at java.util.ArrayList.get(ArrayList.java:308)
       at com.mikepenz.fastadapter.adapters.ItemAdapter.getAdapterItem(ItemAdapter.java:190)
       at com.mikepenz.fastadapter.FastAdapter.getItem(FastAdapter.java:770)
       at com.mikepenz.fastadapter.FastAdapter.getItemViewType(FastAdapter.java:820)
       at com.mikepenz.fastadapter.AbstractAdapter.getItemViewType(AbstractAdapter.java:76)
       at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5631)
       at android.support.v7.widget.GapWorker.prefetchPositionWithDeadline(GapWorker.java:285)
       at android.support.v7.widget.GapWorker.flushTaskWithDeadline(GapWorker.java:342)
       at android.support.v7.widget.GapWorker.flushTasksWithDeadline(GapWorker.java:358)
       at android.support.v7.widget.GapWorker.prefetch(GapWorker.java:365)
       at android.support.v7.widget.GapWorker.run(GapWorker.java:396)
       at android.os.Handler.handleCallback(Handler.java:815)
       at android.os.Handler.dispatchMessage(Handler.java:104)
       at android.os.Looper.loop(Looper.java:207)
       at android.app.ActivityThread.main(ActivityThread.java:5896)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:789)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:679)
```

**Comment by mikepenz** [comment]
Please check out: https://github.com/mikepenz/FastAdapter/pull/594 and see if it solves your issue

**Comment by DummyCo** [comment]
Trying to test as a submodule but getting strange runtime exception :)
Do we have some kind of nightly library snapshots on cloud repositories?

> java.lang.NoSuchMethodError: No virtual method getItemList()Lcom/mikepenz/fastadapter/IItemList; in class Lcom/mikepenz/fastadapter/adapters/ModelAdapter; or its super classes (declaration of 'com.mikepenz.fastadapter.adapters.ModelAdapter' appears in /data/app/TMMPsDvosJFW04P_f-1Lpg==/split_lib_dependencies_apk.apk)

Will try to create separate libraries

**Comment by mikepenz** [comment]
@DummyCo oh sorry. #594 already is updated for the `IItemList` which got not released. both changes together will be released in v3.1

**Comment by DummyCo** [comment]
@mikepenz oh I see. Looking forward to it, thanks!
